plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/LocalDockerSinglePortServerEvaluationStrategy.java

```diff
@@ -41,57 +33,13 @@
  * @author Mario Loriedo <mloriedo@redhat.com>
  * @see ServerEvaluationStrategy
  */
-public class LocalDockerSinglePortServerEvaluationStrategy extends LocalDockerServerEvaluationStrategy {
-
-    private static final String CHE_WORKSPACE_ID_ENV_VAR = "CHE_WORKSPACE_ID";
-
-    private boolean secureExternalUrls;
+public class LocalDockerSinglePortServerEvaluationStrategy extends CustomServerEvaluationStrategy {
 
     @Inject
     public LocalDockerSinglePortServerEvaluationStrategy(@Nullable @Named("che.docker.ip") String internalAddress,
                                                          @Nullable @Named("che.docker.ip.external") String externalAddress,
-                                                         @Named("che.docker.server_evaluation_strategy.secure.external.urls") boolean secureExternalUrls) {
-        super(internalAddress, externalAddress);
-        this.secureExternalUrls = secureExternalUrls;
-    }
-
-    @Override
-    protected boolean useHttpsForExternalUrls() {
-        return this.secureExternalUrls;
-    }
-
-    @Override
-    protected Map<String, String> getExternalAddressesAndPorts(ContainerInfo containerInfo,
-                                                               String internalHost) {
-        String cheExternalAddress = getCheExternalAddress(containerInfo, internalHost);
-        String workspaceID        = getWorkspaceID(containerInfo.getConfig().getEnv());
-        Map<String, String> labels= containerInfo.getConfig().getLabels();
-
-        Map<String, List<PortBinding>> portBindings = containerInfo.getNetworkSettings().getPorts();
-        Map<String, String> addressesAndPorts = new HashMap<>();
-        for (String serverKey : portBindings.keySet()) {
-            String serverName = getWorkspaceServerName(labels, serverKey);
-            String serverURL = serverName + "-" + workspaceID + "-" + cheExternalAddress;
-            addressesAndPorts.put(serverKey, serverURL);
-        }
-
-        return addressesAndPorts;
-    }
-
-    private String getWorkspaceServerName(Map<String, String> labels, String portKey) {
-        ServerConfImpl serverConf = getServerConfImpl(portKey, labels, new HashMap<>());
-        if (serverConf == null) {
-            return "server-" + portKey.split("/",2)[0];
```

Comment:
OK these lines have been removed, how do we deal with exposed ports without label ? we did a quick fix here https://github.com/eclipse/che/pull/4970 while waiting for the rebase to work on a refactored and cleanner fix. Would preview urls works with non label exposed ports ?